components/tutorial/tutorial_tip/tutorial_tip.tsx

```diff
@@ -63,6 +63,32 @@ export default class TutorialTip extends React.PureComponent<Props, State> {
         this.setState({show: false});
     }
 
+    private dismiss = (e: React.MouseEvent | React.KeyboardEvent): void => {
+        this.hide();
+        const wasEscapeDismissal = e.type === 'keyup';
+        const tag = this.props.telemetryTag + '_dismiss';
+        if (wasEscapeDismissal) {
+            trackEvent('tutorial', tag);
+            return;
+        }
+        const target = e.target as HTMLElement;
+
+        // We defer checking this until here because `.closest` is mildly expensive.
+        const wasOutsideClick = (
+
+            // If a user click on an element inside triggers a click outside,
+            // then react-bootstrap's onHide fires onHide for both elements.
+            // Even if the inside click is for opt out, that is already covered in the _skip event
+            !target.closest('.tip-overlay') &&
+
+            // Clicking the post textbox means a prefilled message was chosen, so it is not a dismissal.
+            target.id !== 'post_textbox'
+        );
+        if (wasOutsideClick) {
+            trackEvent('tutorial', tag);
```

Comment:
Per conversation with @esethna we want to track when the user hides the modal. I was hoping this function could be simpler, but for cases like prewritten messages where we need to generate a click to hide the overlay, this function gets called twice, once for the original element click, and once for the create post textbox being clicked/focused. Also, this function gets called when the user clicks "got it" or opts out of the tips. So I had to do some event sniffing to tell if the user was really dismissing the overlay.